File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums)
        n = len(nums)
        a = [[] for _ in range(n)]
        for i in freq:
            a[freq[i] - 1].append(i)
        ans = []
        for i in range(n - 1, -1, -1):
            if a[i]:
                for j in a[i]:
                    ans.append(j)
                    k -= 1
            if k == 0:
                return ans
        return ans

File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int i = 0, n = nums.size();
        while (i < n){
            if (1 <= nums[i] && nums[i] <= n){
                int idx = nums[i] - 1;
                if (nums[i] != nums[idx]){
                    swap(nums[idx], nums[i]);
                    continue;
                }
            }
            i++;
        }
        for (int j = 0 ; j < n ; j++){
            if (nums[j] != j + 1){
                return j + 1;
            }
        }
        return n + 1;
    }
};